crates/turbopack-ecmascript/src/references/esm/url.rs

```diff
@@ -149,177 +113,71 @@ impl CodeGenerateable for UrlAssetReference {
+                            if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(0) {
+                                *expr = quote!(
+                                    "__turbopack_require__($id)" as Expr,
+                                    id: Expr = module_id_to_lit(&id),
+                                );
+                            }
 
-            let ast_path = this.ast_path.await?;
-            visitors.push(
-                create_visitor!(ast_path, visit_mut_expr(new_expr: &mut Expr) {
-                    if let Expr::New(NewExpr { args: Some(args), .. }) = new_expr {
-                        if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(0) {
-                            *expr = quote!(
-                                "__turbopack_require__($id)" as Expr,
-                                id: Expr = module_id_to_lit(&id),
-                            );
+                            if is_browser {
+                                if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(1) {
+                                    *expr = quote!("location.href" as Expr);
+                                }
+                            }
                         }
+                    }),
+                );
+            }
+            ReferencedAsset::OriginalReferenceTypeExternal(request) => {
+                // Handle new URL that points to an external URL
+
+                // In Browser environments, we rewrite the `import.meta.url` to be a
+                // location.origin because it allows us to access files from the
+                // root of the dev server. In node env, the `import.meta.url` already be the correct `file://` URL to load files.
+                let is_browser = *this.is_browser.await?;
+
+                let request = request.to_string();
+                let ast_path = this.ast_path.await?;
+                visitors.push(
+                    create_visitor!(ast_path, visit_mut_expr(new_expr: &mut Expr) {
+                        if let Expr::New(NewExpr { args: Some(args), .. }) = new_expr {
+                            if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(0) {
+                                *expr = Expr::Lit(Lit::Str(Str {
+                                    span: DUMMY_SP,
+                                    value: request.as_str().into(),
+                                    raw: None
+                                }))
+                            }
 
-                        if is_browser {
-                            if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(1) {
-                                *expr = quote!("location.href" as Expr);
+                            if is_browser {
+                                if let Some(ExprOrSpread { box expr, spread: None }) = args.get_mut(1) {
+                                    *expr = quote!("location.href" as Expr);
```

Comment:
This'll also need a update
```suggestion
                                    *expr = quote!("location.origin" as Expr);
```